src/core.rs

```diff
@@ -183,8 +997,30 @@ impl error::Error for ResourceError {}
 trait ResourceTrait {
     fn price_per_hour(&self) -> Result<f32, ResourceError>;
     fn compute(&mut self) -> Result<(), ResourceError>;
+    fn gauge_hour(
+        &self,
+        osc_cost_version: &String,
+        account_id: &String,
+        read_date_rfc3339: &String,
+        region: &String,
+        resource_id: &String,
+        resource_type: &String,
+    ) -> Result<GenericGauge<AtomicF64>, ResourceError>;
+    fn gauge_month(
+        &self,
+        osc_cost_version: &String,
+        account_id: &String,
+        read_date_rfc3339: &String,
+        region: &String,
+        resource_id: &String,
+        resource_type: &String,
+    ) -> Result<GenericGauge<AtomicF64>, ResourceError>;
 }
 
+trait ResourceMetricsTrait {
+    fn register(&self, registry: Registry) -> Result<Registry, ResourceError>;
+    fn add(&self, price_per_hour: f64, price_per_month: f64) -> Result<(), ResourceError>;
```

Comment:
Do we need two functions as we are only fetching one time the resource ?

Reply:
Ok function add remove